core/src/test/scala/org/scalacheck/ops/time/GenericDateTimeGeneratorsSpec.scala

```diff
@@ -5,78 +5,78 @@ import org.scalacheck.ops._
 import org.scalatest.FlatSpec
 import org.scalatest.prop.GeneratorDrivenPropertyChecks
 
+import org.scalatest.Matchers._
 import scala.reflect.ClassTag
 
-abstract class GenericDateTimeGeneratorsSpec[Generators <: AbstractTimeGenerators](
+private[time] abstract class GenericDateTimeGeneratorsSpec[Generators <: AbstractTimeGenerators](
   protected val gen: Generators
 )(implicit classTag: ClassTag[Generators]) extends FlatSpec
   with GeneratorDrivenPropertyChecks {
 
   protected val genClassName: String = classTag.runtimeClass.getSimpleName.stripSuffix("$")
-  protected def arbDateTimeType: Arbitrary[gen.InstantType]
-  protected def clsTagDateTimeType: ClassTag[gen.InstantType]
+  protected def arbInstantType: Arbitrary[gen.InstantType]
+  protected def clsTagInstantType: ClassTag[gen.InstantType]
+  protected def orderingInstantType: Ordering[gen.InstantType]
 
   // set the local implicits
-  private implicit def localArbDateTimeType = arbDateTimeType
-  private implicit def localClsTagDateTimeType = clsTagDateTimeType
-  private implicit def localParams = gen.defaultParams
+  private implicit def localArbDateTimeType: Arbitrary[gen.InstantType] = arbInstantType
+  private implicit def localClsTagDateTimeType: ClassTag[gen.InstantType] = clsTagInstantType
+  private implicit def localParams: gen.ParamsType = gen.defaultParams
+  private implicit def localOrderingInstantType: Ordering[gen.InstantType] = orderingInstantType
 
   behavior of s"$genClassName.before"
 
   it should "not generate errors" in {
-    forAll() { (start: gen.InstantType) =>
+    forAll() { start: gen.InstantType =>
       val sampleIter = gen.before(start).sampleIterator
       val samples = sampleIter.take(10).toSeq
       assert(samples.forall(_.isDefined))
     }
   }
 
-  it should "always generate DateTimes that are less than the given time" in {
-    forAll() { (start: gen.InstantType) =>
-      val startMillis = gen.millis(start)
-      forAll() { (before: gen.InstantType) =>
-        gen.millis(before) < startMillis
```

Comment:
This used to evaluate to a `Boolean` but it would not fail the test if it evaluated to `false` (and thus the tests were broken)